src/test/java/bio/terra/workspace/service/datareference/DataReferenceServiceTest.java

```diff
@@ -193,6 +159,27 @@ public void testDeleteMissingDataReference() {
             dataReferenceService.deleteDataReference(workspaceId, UUID.randomUUID(), USER_REQUEST));
   }
 
+  /**
+   * Hard code serialized values to check that code changes do not break backwards compatibility of
+   * stored JSON values. If these tests fail, your change may not work with existing databases.
+   */
+  @Test
+  public void SnapshotReferenceSerializationBackwardsCompatible() throws Exception {
+    SnapshotReference snapshotReference =
+        (SnapshotReference)
+            ReferenceObject.fromJson(
+                "{\"instanceName\":\"foo\",\"snapshot\":\"bar\"}",
+                DataReferenceType.DATA_REPO_SNAPSHOT);
+    assertEquals("foo", snapshotReference.instanceName());
+    assertEquals("bar", snapshotReference.snapshot());
+  }
+
+  @Test
+  public void SnapshotReferenceDeserializationnBackwardsCompatible() throws Exception {
+    String serializedSnapshot = SnapshotReference.create("foo", "bar").toJson();
+    assertEquals("{\"instanceName\":\"foo\",\"snapshot\":\"bar\"}", serializedSnapshot);
+  }
```

Comment:
I think these should be on a ReferenceObjectTest or a SnapshotReferenceTest file. Unit tests generally go with a file dedicated to the class they're testing.